<a href="https://colab.research.google.com/github/antussa2016/thesis_work/blob/master/data_analysis/tweet_analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
import pandas as pd
import re
import os

In [ ]:
df = pd.read_csv('/content/drive/MyDrive/Demo_dataframe.csv')

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1247 entries, 0 to 1246
Data columns (total 13 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Twitter ID              1247 non-null   object 
 1   big5_openness           1247 non-null   float64
 2   big5_conscientiousness  1247 non-null   float64
 3   big5_extraversion       1247 non-null   float64
 4   big5_agreeableness      1247 non-null   float64
 5   big5_neuroticism        1247 non-null   float64
 6   class_label             1247 non-null   object 
 7   Reputation              1247 non-null   float64
 8   Question                1247 non-null   float64
 9   Answer                  1247 non-null   float64
 10  Bert_encoding           1247 non-null   object 
 11  Twitter_sername         1247 non-null   object 
 12  New_encoded_values      1247 non-null   object 
dtypes: float64(8), object(5)
memory usage: 126.8+ KB


# **delete duplicates**

In [ ]:
cols_to_check = ['Twitter ID']

# Drop the duplicates
df = df.drop_duplicates(subset=cols_to_check, keep='first')

# Write the cleaned dataset to a new file


In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 805 entries, 0 to 1245
Data columns (total 13 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Twitter ID              805 non-null    object 
 1   big5_openness           805 non-null    float64
 2   big5_conscientiousness  805 non-null    float64
 3   big5_extraversion       805 non-null    float64
 4   big5_agreeableness      805 non-null    float64
 5   big5_neuroticism        805 non-null    float64
 6   class_label             805 non-null    object 
 7   Reputation              805 non-null    float64
 8   Question                805 non-null    float64
 9   Answer                  805 non-null    float64
 10  Bert_encoding           805 non-null    object 
 11  Twitter_sername         805 non-null    object 
 12  New_encoded_values      805 non-null    object 
dtypes: float64(8), object(5)
memory usage: 88.0+ KB


In [ ]:
df1=pd.read_csv('/content/drive/MyDrive/changed_encoded_files.csv')

In [ ]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 963 entries, 0 to 962
Data columns (total 4 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   File name         963 non-null    object
 1   new_column        963 non-null    object
 2   new_name1         963 non-null    object
 3   Extracting_name1  963 non-null    object
dtypes: object(4)
memory usage: 30.2+ KB


In [ ]:
df1['Extracting_name1'] = df1['Extracting_name1'].replace('stienmantweet', 'stienman')

In [ ]:
df1['Extracting_name1'] = df1['Extracting_name1'].replace('bi', 'tweetsbi')

In [ ]:
k = 2
i = 2

new_column = []
for index, row in df.iterrows():
    print("1st loop",i)

    j = 0
    p=0
    for value in df1["Extracting_name1"]:
        if j == 0:
            if row['Twitter_sername'] == value:
                # print( f"{row['Twitter_sername']}",value,k,"main dataframe", i)

                # print("2nd loop",k)
                k+=1

                name1= df1["File name"][p]
                new_column.append(name1)
                j += 1

            else:
                p += 1
                continue
                # print(f"{row2}")
                print("NaN")
        p += 1
    i += 1

df['File name']=new_column

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 805 entries, 0 to 1245
Data columns (total 14 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Twitter ID              805 non-null    object 
 1   big5_openness           805 non-null    float64
 2   big5_conscientiousness  805 non-null    float64
 3   big5_extraversion       805 non-null    float64
 4   big5_agreeableness      805 non-null    float64
 5   big5_neuroticism        805 non-null    float64
 6   class_label             805 non-null    object 
 7   Reputation              805 non-null    float64
 8   Question                805 non-null    float64
 9   Answer                  805 non-null    float64
 10  Bert_encoding           805 non-null    object 
 11  Twitter_sername         805 non-null    object 
 12  New_encoded_values      805 non-null    object 
 13  File name               805 non-null    object 
dtypes: float64(8), object(6)
memory usage: 94

In [ ]:
folder_path = '/content/drive/MyDrive/tweet1/tweet'

In [ ]:
file_names = os.listdir(folder_path)

In [ ]:
csv_files = [file_name for file_name in file_names if file_name.endswith('.csv')]

In [ ]:
import nltk
nltk.download('punkt')
from nltk.tokenize import word_tokenize

[nltk_data] Downloading package punkt to /root/nltk_data...
[nltk_data]   Unzipping tokenizers/punkt.zip.


In [ ]:
nltk.download('stopwords')

[nltk_data] Downloading package stopwords to /root/nltk_data...
[nltk_data]   Unzipping corpora/stopwords.zip.


True

In [ ]:
k = 0
num_words = 0
new_column = []
for index, row in df.iterrows():
    for file_name in csv_files:
        if row['File name'] == file_name:
            df3=file_name
            df3=pd.read_csv(f'/content/drive/MyDrive/tweet1/tweet/{df3}')
            # column=df3['tweet'].name

            if ('tweet' in df3.columns) & ('text' not in df3.columns):
                tweets_column = "tweet"
                tokens = df3[tweets_column].apply(word_tokenize)

                # Flatten the list of tokens
                words = [word for tweet in tokens for word in tweet]

                # Remove stopwords and punctuation marks
                stopwords = set(nltk.corpus.stopwords.words('english'))
                words = [word.lower() for word in words if word.isalpha() and word.lower() not in stopwords]

                # Count the number of words
                num_words = len(words)
                # num_tweets += len(df3[tweets_column])
                print("tweet words",num_words,k)
                new_column.append(num_words)
                k +=1
            else:
                tweets_column = "text"
                df4 = pd.DataFrame(df3[tweets_column].dropna().tolist(),columns=['text'])

                tokens = df4[tweets_column].apply(word_tokenize)

                # Flatten the list of tokens
                words = [word for tweet in tokens for word in tweet]

                # Remove stopwords and punctuation marks
                stopwords = set(nltk.corpus.stopwords.words('english'))
                words = [word.lower() for word in words if word.isalpha() and word.lower() not in stopwords]

                # Count the number of words
                num_words = len(words)

                print("tweet words",num_words,k)
                new_column.append(num_words)
                k +=1




print(num_words)


In [ ]:
df['words'] = new_column

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 805 entries, 0 to 1245
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Twitter ID              805 non-null    object 
 1   big5_openness           805 non-null    float64
 2   big5_conscientiousness  805 non-null    float64
 3   big5_extraversion       805 non-null    float64
 4   big5_agreeableness      805 non-null    float64
 5   big5_neuroticism        805 non-null    float64
 6   class_label             805 non-null    object 
 7   Reputation              805 non-null    float64
 8   Question                805 non-null    float64
 9   Answer                  805 non-null    float64
 10  Bert_encoding           805 non-null    object 
 11  Twitter_sername         805 non-null    object 
 12  New_encoded_values      805 non-null    object 
 13  File name               805 non-null    object 
 14  words                   805 non-null    i

In [ ]:

max_tweets =df['words'].min()
max_tweets

68

In [ ]:
k = 0
num_words = 0
new_column = []

for index, row in df.iterrows():
    for file_name in csv_files:
        if row['File name'] == file_name:
            df3=file_name
            df3=pd.read_csv(f'/content/drive/MyDrive/tweet1/tweet/{df3}')
            # column=df3['tweet'].name

            if ('tweet' in df3.columns) & ('text' not in df3.columns):
                tweets_column = "tweet"
                num_words = len(df3[tweets_column])
                # num_tweets += len(df3[tweets_column])
                print("total tweet", num_words,k)
                new_column.append(num_words)
                k +=1
            else:
                tweets_column = "text"

                num_words = len(df3[tweets_column].dropna().tolist())
                new_column.append(num_words)
                print("total tweet", num_words,k)
                k +=1




avg_tweets = num_words / df['File name'].nunique()
print("total and averge tweets",num_words,avg_tweets)


In [ ]:
df['tweet_number'] = new_column

In [ ]:
max_tweets =df['tweet_number'].max()
print(max_tweets)
min_tweets =df['tweet_number'].min()
print(min_tweets)

3247
9


In [ ]:
k = 0
num_words = 0
new_column = []
total_words = 0

for index, row in df.iterrows():
    for file_name in csv_files:
        if row['File name'] == file_name:
            df3=file_name
            df3=pd.read_csv(f'/content/drive/MyDrive/tweet1/tweet/{df3}')
            # column=df3['tweet'].name

            if ('tweet' in df3.columns) & ('text' not in df3.columns):
                tweets_column = "tweet"
                df3['num_words'] = df3[tweets_column].apply(lambda x: len(x.split()))
                total_words += df3['num_words'].sum()
                # num_tweets += len(df3[tweets_column])
                print("total tweet", total_words,k)

                k +=1
            else:
                tweets_column = "text"
                df4 = pd.DataFrame(df3[tweets_column].dropna().tolist(),columns=['text'])
                df4['num_words'] = df4[tweets_column].apply(lambda x: len(x.split()))
                total_words += df4['num_words'].sum()

                print("total tweet", total_words,k)
                k +=1

print ("total sum of words from tweets",total_words)
print ("Average of words in tweets",total_words/805)


In [ ]:
21211409/805

26349.576397515528

# **Result**



# **Total tweet =  1477924**
average tweets = 1840.503113325031

max number of  tweet of a user = 3247

min number of  tweet of a user =9

# **Total words in tweets = 12710448**

avera of words in tweets of a user = 27015.201242236024

max number of words in tweet of a user = 42123

min number of words in tweet of a user = 68